### PR TITLE
Semver tag release candidate lifecycle images

### PR DESCRIPTION
**Pull request by natalieparellano**
As discussed in implementation sync. We'll semver tag the images but NOT update `latest`.

Signed-off-by: Natalie Arellano <narellano@vmware.com>

### CODE REVIEW

**Comment by natalieparellano** [comment]
Tested on fork:

https://github.com/natalieparellano/lifecycle/actions/runs/962261399 produced https://github.com/natalieparellano/lifecycle/releases/tag/v0.44.0 and was followed by https://github.com/natalieparellano/lifecycle/actions/runs/962263306

https://github.com/natalieparellano/lifecycle/actions/runs/962314609 produced https://github.com/natalieparellano/lifecycle/releases/tag/v0.55.0-rc.1 and was followed by https://github.com/natalieparellano/lifecycle/actions/runs/962316503

File: .github/workflows/draft-release.yml [hunk 1]
```diff
@@ -112,16 +109,17 @@ jobs:
           lifecycle_path=$(ls artifact-windows/lifecycle-*windows.x86-64.tgz)
           echo "ARTIFACT_WINDOWS_PATH=$PWD/$lifecycle_path" >> $GITHUB_ENV
       - name: Create Pre Release
-        if: contains(env.LIFECYCLE_VERSION, 'rc')
+        if: contains(env.LIFECYCLE_VERSION, 'rc') # e.g., 0.99.0-rc.1
         id: create_pre_release
-        uses: actions/create-release@latest
+        uses: softprops/action-gh-release@v1
```
Comment (jabrown85) [inline .github/workflows/draft-release.yml:114]: Have we used this action elsewhere? Do we need to vet any new actions?

File: .github/workflows/post-release.yml [hunk 2]
```diff
@@ -3,58 +3,104 @@ name: post-release
...
           echo "LIFECYCLE_VERSION=$(echo ${{ github.event.release.tag_name }} | cut -d "v" -f2)" >> $GITHUB_ENV
           echo "LIFECYCLE_IMAGE_TAG=$(git describe --always --dirty)" >> $GITHUB_ENV
-      - name: Retag release candidate lifecycle images
+      - name: Retag release candidate lifecycle images - semver
         run: |
           docker pull buildpacksio/lifecycle:${{ env.LIFECYCLE_IMAGE_TAG }}-linux
           docker image tag buildpacksio/lifecycle:${{ env.LIFECYCLE_IMAGE_TAG }}-linux buildpacksio/lifecycle:${{ env.LIFECYCLE_VERSION }}-linux
-          docker image tag buildpacksio/lifecycle:${{ env.LIFECYCLE_IMAGE_TAG }}-linux buildpacksio/lifecycle:latest-linux
           docker push buildpacksio/lifecycle:${{ env.LIFECYCLE_VERSION }}-linux
+      - name: Retag release candidate lifecycle images - latest
```
Comment (jabrown85) [inline .github/workflows/post-release.yml:42]:
```suggestion
      - name: Retag lifecycle images - latest
```

File: .github/workflows/post-release.yml [hunk 3]
```diff
@@ -3,58 +3,104 @@ name: post-release
...
           echo "LIFECYCLE_VERSION=$(echo ${{ github.event.release.tag_name }} | cut -d "v" -f2)" >> $env:GITHUB_ENV
           echo "LIFECYCLE_IMAGE_TAG=$(git describe --always --dirty)" >> $env:GITHUB_ENV
-      - name: Retag release candidate lifecycle images
+      - name: Retag release candidate lifecycle images - semver
         run: |
           docker pull buildpacksio/lifecycle:${{ env.LIFECYCLE_IMAGE_TAG }}-windows
           docker image tag buildpacksio/lifecycle:${{ env.LIFECYCLE_IMAGE_TAG }}-windows buildpacksio/lifecycle:${{ env.LIFECYCLE_VERSION }}-windows
-          docker image tag buildpacksio/lifecycle:${{ env.LIFECYCLE_IMAGE_TAG }}-windows buildpacksio/lifecycle:latest-windows
           docker push buildpacksio/lifecycle:${{ env.LIFECYCLE_VERSION }}-windows
+      - name: Retag release candidate lifecycle images - latest
```
Comment (jabrown85) [inline .github/workflows/post-release.yml:81]:
```suggestion
      - name: Retag lifecycle images - latest
```

File: .github/workflows/draft-release.yml (hunk 1 above)
Comment (natalieparellano) [reply .github/workflows/draft-release.yml:114]: It's referred to by https://github.com/actions/create-release and used in `pack`, but I discovered some issues in how we were using it (see https://github.com/buildpacks/pack/issues/1213) and from [this thread](https://github.com/softprops/action-gh-release/issues/107) I decided to switch to using `gh` cli. Now commits coming soon...

**Comment by natalieparellano** [comment]
Tested on fork:

https://github.com/natalieparellano/lifecycle/actions/runs/965766740 produced https://github.com/natalieparellano/lifecycle/releases/tag/v0.77.1 followed by https://github.com/natalieparellano/lifecycle/actions/runs/965767966
```
$  docker run natalieparellano/lifecycle:0.77.1 /cnb/lifecycle/lifecycle -version
0.77.1+f13d054
```

https://github.com/natalieparellano/lifecycle/actions/runs/965128058 produced https://github.com/natalieparellano/lifecycle/releases/tag/v0.78.0-rc.1 followed by https://github.com/natalieparellano/lifecycle/actions/runs/965141327
```
$  docker run natalieparellano/lifecycle:0.78.0-rc.1 /cnb/lifecycle/lifecycle -version
0.78.0-rc.1+b8a143c
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -109,10 +109,13 @@ jobs:
         with:
           name: lifecycle-windows-x86-64
           path: out/lifecycle-v*+windows.x86-64.tgz
+      - uses: azure/docker-login@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (natalieparellano) [inline .github/workflows/build.yml:115]: I don't like seeing `WARNING! Your password will be stored unencrypted in /home/runner/.docker/config.json.` But wondering how we feel about introducing a new dependency - @jabrown85 WDYT?

File: .github/workflows/draft-release.yml [hunk 5]
```diff
@@ -111,97 +108,51 @@ jobs:
...
-            # lifecycle v${{ env.LIFECYCLE_VERSION }}
+        run: |
+          echo "RELEASE_KIND=release" >> $GITHUB_ENV
+      - name: Set release body text
+        run: |
+          cat << EOF > body.txt
+          # lifecycle v${{ env.LIFECYCLE_VERSION }}
 
-            Welcome to `v${{ env.LIFECYCLE_VERSION }}`, a **beta** release of the Cloud Native Buildpacks Lifecycle.
+          Welcome to v${{ env.LIFECYCLE_VERSION }}, a **beta** ${{ env.RELEASE_KIND }} of the Cloud Native Buildpacks Lifecycle.
```
Comment (natalieparellano) [inline .github/workflows/draft-release.yml:124]: Sadly this means we'll have to manually add backticks if we want interpolation. If there's a better way to do this in bash, I'd love to hear about it!

File: .github/workflows/build.yml (hunk 4 above)
Comment (jabrown85) [reply .github/workflows/build.yml:115]: MS already owns github so I feel 🆗  on this new action.

**Review by jabrown85 — APPROVED** [review]

**Pull request merged**